src/governance-vote.cpp

```diff
@@ -35,108 +35,205 @@ std::string CGovernanceVoting::ConvertOutcomeToString(int nOutcome)
+            break;
+        case VOTE_SIGNAL_CUSTOM15:
+            strReturn = "CUSTOM15";
+            break;
+        case VOTE_SIGNAL_CUSTOM16:
+            strReturn = "CUSTOM16";
+            break;
+        case VOTE_SIGNAL_CUSTOM17:
+            strReturn = "CUSTOM17";
+            break;
+        case VOTE_SIGNAL_CUSTOM18:
+            strReturn = "CUSTOM18";
+            break;
+        case VOTE_SIGNAL_CUSTOM19:
+            strReturn = "CUSTOM19";
+            break;
+        case VOTE_SIGNAL_CUSTOM20:
+            strReturn = "CUSTOM20";
+            break;
     }
 
-    return "error";
+    return strReturn;
 }
 
 
-int CGovernanceVoting::ConvertVoteOutcome(std::string strVoteOutcome)
+vote_outcome_enum_t CGovernanceVoting::ConvertVoteOutcome(std::string strVoteOutcome)
 {
-    int nVote = -1;
-    if(strVoteOutcome == "yes") nVote = VOTE_OUTCOME_YES;
-    if(strVoteOutcome == "no") nVote = VOTE_OUTCOME_NO;
-    if(strVoteOutcome == "abstain") nVote = VOTE_OUTCOME_ABSTAIN;
-    if(strVoteOutcome == "none") nVote = VOTE_OUTCOME_NONE;
-    return nVote;
+    vote_outcome_enum_t eVote = VOTE_OUTCOME_NONE;
+    if(strVoteOutcome == "yes") {
+        eVote = VOTE_OUTCOME_YES;
+    }
+    else if(strVoteOutcome == "no") {
+        eVote = VOTE_OUTCOME_NO;
+    }
+    else if(strVoteOutcome == "abstain") {
+        eVote = VOTE_OUTCOME_ABSTAIN;
+    }
+    else if(strVoteOutcome == "none") {
+        eVote = VOTE_OUTCOME_NONE;
+    }
+    return eVote;
 }
 
-int CGovernanceVoting::ConvertVoteSignal(std::string strVoteSignal)
+vote_signal_enum_t CGovernanceVoting::ConvertVoteSignal(std::string strVoteSignal)
 {
-    if(strVoteSignal == "none") return VOTE_SIGNAL_NONE;         // 0
-    if(strVoteSignal == "funding") return VOTE_SIGNAL_FUNDING;   // 1
-    if(strVoteSignal == "valid") return VOTE_SIGNAL_VALID;       // 2
-    if(strVoteSignal == "delete") return VOTE_SIGNAL_DELETE;     // 3
-    if(strVoteSignal == "endorsed") return VOTE_SIGNAL_ENDORSED; // 4
+    vote_signal_enum_t eSignal = VOTE_SIGNAL_NONE;
+    if(strVoteSignal == "none") {
+        eSignal = VOTE_SIGNAL_NONE;  
+    }
+    else if(strVoteSignal == "funding") {
+        eSignal = VOTE_SIGNAL_FUNDING;
+    }
+    else if(strVoteSignal == "valid") {
+        eSignal = VOTE_SIGNAL_VALID;
+    }
+    if(strVoteSignal == "delete") {
+        eSignal = VOTE_SIGNAL_DELETE;
+    }
+    if(strVoteSignal == "endorsed") {
+        eSignal = VOTE_SIGNAL_ENDORSED;
+    }
 
     // ID FIVE THROUGH CUSTOM_START ARE TO BE USED BY GOVERNANCE ENGINE / TRIGGER SYSTEM
 
     // convert custom sentinel outcomes to integer and store
     try {
         int  i = boost::lexical_cast<int>(strVoteSignal);
-        if(i < VOTE_SIGNAL_CUSTOM_START || i > VOTE_SIGNAL_CUSTOM_END) return -1;
-        return i;
+        if(i < VOTE_SIGNAL_CUSTOM1 || i > VOTE_SIGNAL_CUSTOM20) {
+            eSignal = VOTE_SIGNAL_NONE;
+        }
+        else  {
+            eSignal = vote_signal_enum_t(i);
```

Comment:
Holy crap... Now I finally get the idea Evan had behind that, thanks! The only thing left for me to understand now is why we are declaring NOOPs and custom actions we do not use in any way, but I guess that's out of the scope of this PR :)